src/Tests/dotnet-workload-install.Tests/WorkloadGarbageCollectionTests.cs

```diff
@@ -0,0 +1,393 @@
+
+        //  Additional scenarios to add tests for once workload sets are added:
+        //  Garbage collect workload sets
+        //  Garbage collect with install state with workload set
+        //  Don't garbage collect baseline workload set
+
+        void PackShouldExist(PackInfo pack, bool shouldExist)
+        {
+            if (pack.Kind == WorkloadPackKind.Library)
+            {
+                if (shouldExist)
+                {
+                    new FileInfo(pack.Path).Should().Exist();
+                }
+                else
+                {
+                    new FileInfo(pack.Path).Should().NotExist();
+                }
+            }
+            else
+            {
+                if (shouldExist)
+                {
+                    new DirectoryInfo(pack.Path).Should().Exist();
+                }
+                else
+                {
+                    new DirectoryInfo(pack.Path).Should().NotExist();
+                }
+            }
+        }
+
+        PackInfo CreatePackInfo(string id, string version, WorkloadPackKind kind, string resolvedPackageId = null)
+        {
+            if (resolvedPackageId == null)
+            {
+                resolvedPackageId = id;
+            }
+
+            string path;
+            if (kind == WorkloadPackKind.Library)
+            {
+                path = Path.Combine(_dotnetRoot, "library-packs", $"resolvedPackageId.{version}.nupkg");
+            }
+            else
+            {
+                path = Path.Combine(_dotnetRoot, "packs", id, version);
+            }
+
+            return new PackInfo(new WorkloadPackId(id), version, kind, path, resolvedPackageId);
+        }
+
+        FileInfo PackRecord(PackInfo pack, string sdkFeatureBand)
+        {
+            var installedPacksPath = Path.Combine(_dotnetRoot, "metadata", "workloads", "InstalledPacks", "v1");
+            var packRecordPath = Path.Combine(installedPacksPath, pack.Id, pack.Version, sdkFeatureBand);
+            return new FileInfo(packRecordPath);
+        }
+
+
+        private void CreateInstalledPack(PackInfo pack, string sdkFeatureBand)
+        {
+            //  Write pack installation record
+            var packRecordPath = PackRecord(pack, sdkFeatureBand);
+            Directory.CreateDirectory(Path.GetDirectoryName(packRecordPath.FullName));
+            var packRecordContents = JsonSerializer.Serialize<WorkloadResolver.PackInfo>(pack);
+            File.WriteAllText(packRecordPath.FullName, packRecordContents);
+
+            //  Create fake pack install
+            if (pack.Kind == WorkloadPackKind.Library)
+            {
+                Directory.CreateDirectory(Path.GetDirectoryName(pack.Path));
+                using var _ = File.Create(pack.Path);
+            }
+            else
+            {
+                Directory.CreateDirectory(pack.Path);
+            }
+        }
+
+        private void CreateInstallState(string featureBand, string installStateContents)
```

Comment:
Some of these functions have been duplicated a lot across tests, such as this one. There is a function `CreateMockInstallState` in `SdkDirectoryWorkloadManifestProviderTests` which is nearly identical, however, it does not use the featureBand parameter correctly. I do understand the simplicity of using these functions per test file, though I think it would be best for maintenance's sake to try to coalesce them into one `WorkloadTestUtilities.cs` file

Reply:
Yes, I copied this from `SdkDirectoryWorkloadManifestProviderTests`, and noticed the bug so I fixed it in this PR.

I think most of the helper methods use private fields of the class so that they're less verbose to call.  So it's not so simple to factor them out into a shared file, I think you'd need a base class for workload tests.  I'd be happy to see someone try that, but I didn't want to tackle that here.

Overall, I don't like duplicated code, but I think I'm a lot more OK with it in tests.

Reply:
Ah, I did not even notice that you had fixed `CreateMockInstallState`. I would also be happy to have some of this compressed down together, though it doesn't have to be in this PR. I will give it a 👍, though I would be more pleased once these get cleaned up in the future ;)